debbuild

```diff
@@ -1485,8 +1622,15 @@ sub binpackage {
     foreach my $scr (qw(pre post preun postun)) {
       my $scrfile = $scr;
       $scrfile .= 'inst' unless $scrfile =~ s/un/rm/;
-      if ($pkgdata{$pkg}{$scr}) {
-        (my $content = expandmacros($pkgdata{$pkg}{$scr})) =~ s/^[\s]*$//mg;
+      if ($pkgdata{$pkg}{$scr} || (($scr eq 'pre') && defined($pkgdata{$pkg}{pretrans})) ||
+                                  (($scr eq 'post') && defined($pkgdata{$pkg}{posttrans}))) {
+        my $content = defined($pkgdata{$pkg}{$scr}) ? expandmacros($pkgdata{$pkg}{$scr}) : '';
+        $content = expandmacros($pkgdata{$pkg}{pretrans})."\n".$content
+                   if ($scr eq 'pre' and defined($pkgdata{$pkg}{pretrans}));
+        $content .= expandmacros($pkgdata{$pkg}{posttrans})
+                   if ($scr eq 'post' and defined($pkgdata{$pkg}{posttrans}));
+        $content =~ s/^[\s]*$//mg;
+        $content = ($content ? $content : '').$fmo if $scr eq 'post' and $fmo ne '';
```

Comment:
Please add warnings for when `pretrans` and `posttrans` are coerced to `pre` and `post`. The semantics of `pretrans` and `posttrans` don't exist in dpkg and such coercing can lead to broken packages in some cases, so making sure it's recorded in build logs that it happened can help with debugging.

We already do this when we coerce `Supplements` to `Enhances` and similar.

Reply:
Is the warning message like this is ok for such cases?:
```
Warning: 'pretrans/posttans' script is not natively supported by .deb packages.
Merging with 'inst' script.
```

Reply:
Something like that:

```
Warning: '$type' script is not natively supported by .deb packages.
Merging with '$othertype' script.
```

Being specific matters here.

Reply:
warnings appended